pkg/appfile/parser.go

```diff
@@ -554,3 +602,133 @@ func PrepareProcessContext(k8sClient client.Client, wl *Workload, applicationNam
+			if err != nil {
+				return nil, err
+			}
+
+			requiredSecret = append(requiredSecret, process.RequiredSecrets{
+				Name:        secretName,
+				ContextName: contextName,
+				Namespace:   namespace,
+				Data:        secretData,
+			})
+		}
+	}
+	return requiredSecret, nil
+}
+
+func extractSecret(ctx context.Context, c client.Client, namespace, name string) (map[string]interface{}, error) {
+	secretData := make(map[string]interface{})
+	var secret v1.Secret
+	if err := c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &secret); err != nil {
+		return nil, fmt.Errorf("failed to get secret %s from namespace %s which is required by the component: %w",
+			name, namespace, err)
+	}
+	for k, v := range secret.Data {
+		secretData[k] = string(v)
+	}
+	if len(secretData) == 0 {
+		return nil, fmt.Errorf("data in secret %s from namespace %s isn't available", name, namespace)
+	}
+	return secretData, nil
+}
+
+func getComponentSetting(settingParamName string, params map[string]interface{}) (interface{}, error) {
+	if secretName, ok := params[settingParamName]; ok {
+		return secretName, nil
+	}
+	return nil, fmt.Errorf("failed to get the value of component setting %s", settingParamName)
+}
+
+// IsCloudResourceProducer checks whether a workload is cloud resource producer role
+func (wl *Workload) IsCloudResourceProducer() bool {
+	var existed bool
+	_, existed = wl.Params[process.OutputSecretName]
+	return existed
+}
+
+// IsCloudResourceConsumer checks whether a workload is cloud resource consumer role
+func (wl *Workload) IsCloudResourceConsumer() bool {
+	requiredSecretTag := strings.TrimRight(utils.InsertSecretToTag, "=")
+	matched, err := regexp.Match(regexp.QuoteMeta(requiredSecretTag), []byte(wl.Template))
+	if err != nil || !matched {
+		return false
+	}
+	return true
+}
+
+func parseWorkloadEnvMapping(ctx context.Context, c client.Client, namespace string, envMappings map[string]v1beta1.EnvMapping) ([]v1.EnvVar, error) {
+	var envs []v1.EnvVar
+	for envName, v := range envMappings {
+		// Secret from another component could not be checked, or the whole application will fail, which means we could
+		// not support secret value which contain variables.
+		// secretData, err := extractSecret(ctx, c, namespace, v.Secret)
+		//if err != nil {
+		//	return nil, err
+		//}
+		if v.Value != "" {
+			//TODO(zzxwill) need to find a way to extract `value: oss-conn.bucket + "/" + oss-conn.endpoint`
+			envs = append(envs, v1.EnvVar{Name: envName, Value: v.Value})
+			continue
+		}
+		var secretKey string
+		if v.Key != "" {
+			secretKey = v.Key
+		} else {
+			secretKey = envName
+		}
+
+		//secretValue, ok := secretData[secretKey]
+		//if !ok {
+		//	return nil, fmt.Errorf("failed to retrieve key %s from secret %s in namespace %s", v.Key, v.Secret, namespace)
+		//}
+		//secretValueStr, ok := secretValue.(string)
+		//if !ok {
+		//	return nil, fmt.Errorf("failed to convert %s to string type", secretValue)
+		//}
```

Comment:
What these commented out code  use for?

Reply:
Tried to discuss in the morning for getting secret value issue. Removed.